src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,161 @@
-
+        if (size > capacity * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = hash(key, capacity);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        Node<K, V> node = table[index];
+        if (node == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            while (node.next != null || checkKey(key, node)) {
+                if (checkKey(key, node)) {
+                    node.value = value;
+                    return;
+                }
+                node = node.next;
+            }
+            node.next = newNode;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key, capacity);
+        Node<K, V> node = table[index];
+        if (node == null) {
+            return null;
+        }
+        while (node != null) {
+            if (checkKey(key, node)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V> transfer(Node<K, V> node, Node<K, V>[] newTable) {
+        Node<K, V> tempNode = null;
+        if (node.next != null) {
+            tempNode = node.next;
+        }
+        node.next = null;
+        int index = hash(node.key, capacity);
+        if (newTable[index] == null) {
+            newTable[index] = node;
+        } else {
+            Node<K, V> nextNode = newTable[index].next;
+            Node<K, V> lastNode = newTable[index];
+            while (nextNode != null) {
+                if (nextNode.next == null) {
+                    lastNode = nextNode;
+                    break;
+                }
+                nextNode = nextNode.next;
+            }
+            lastNode.next = node;
+        }
+        return tempNode;
+    }
+
+    private void resize() {
+        int newCapacity = capacity * 2;
+        capacity = newCapacity;
+        Node<K, V>[] newTable = new Node[newCapacity];
+        Node<K, V> nextNode = null;
+        for (Node<K, V> node : table) {
+            if (node == null) {
+                continue;
+            }
+            nextNode = transfer(node, newTable);
+
+            while (nextNode != null) {
+                nextNode = transfer(nextNode, newTable);
+            }
+        }
+        table = newTable;
+    }
+
+    private boolean checkKey(K key, Node<K, V> node) {
+        if (node.key == key || node.key != null && node.key.equals(key)) {
+            return true;
+        }
+        return false;
+    }
+
+    static class Node<K, V> implements Map.Entry<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        @Override
+        public K getKey() {
```

Comment:
[Don't use getters and setters in class Node](https://github.com/mate-academy/jv-hash-map/blob/master/checklist.md#dont-use-getters-and-setters-in-class-node)

Reply:
I understand, but if I remove it than I see error that getter and setter should be or make this class abstract 
![Screenshot_6](https://github.com/user-attachments/assets/847a4311-a47a-48a5-849d-c95ac47e6054)